File: 2026-04-April-GFG-POTD/April 16 - Implement Atoi/Solution.py
class Solution:
    def myAtoi(self, s: str) -> int:
        i, n = 0, len(s)
        
        # 1. Skip leading whitespaces
        while i < n and s[i] == ' ':
            i += 1
        
        if i == n:
            return 0
        
        # 2. Check sign
        sign = 1
        if s[i] in ['+', '-']:
            sign = -1 if s[i] == '-' else 1
            i += 1
        
        # 3. Read digits
        result = 0
        
        while i < n and s[i].isdigit():
            digit = int(s[i])
            result = result * 10 + digit
            
            # 4. Handle overflow
            if sign * result > 2**31 - 1:
                return 2**31 - 1
            if sign * result < -2**31:
                return -2**31
            
            i += 1
        
        return sign * result

File: 2026-04-April-GFG-POTD/April 16 - Implement Atoi/Solution.cpp
class Solution {
public:
    int myAtoi(string s) {
        int i = 0, n = s.size();
        
        // 1. Skip leading whitespaces
        while (i < n && s[i] == ' ') {
            i++;
        }
        
        if (i == n) return 0;
        
        // 2. Check sign
        int sign = 1;
        if (s[i] == '+' || s[i] == '-') {
            sign = (s[i] == '-') ? -1 : 1;
            i++;
        }
        
        // 3. Read digits
        long long result = 0;
        
        while (i < n && isdigit(s[i])) {
            int digit = s[i] - '0';
            
            result = result * 10 + digit;
            
            // 4. Handle overflow
            if (sign * result > INT_MAX) return INT_MAX;
            if (sign * result < INT_MIN) return INT_MIN;
            
            i++;
        }
        
        return (int)(sign * result);
    }
